app/src/main/java/org/eyeseetea/malariacare/data/database/model/Survey.java

```diff
@@ -311,6 +314,21 @@ public void setEventUid(String eventuid) {
     }
 
     /**
+     * Ratio of completion is cached into answeredQuestionRatio in order to speed up loading
+     */
+    public SurveyAnsweredRatio getAnsweredQuestionRatio() {
+        if (answeredQuestionRatio == null) {
+            answeredQuestionRatio = SurveyAnsweredRatioCache.get(this.getId_survey());
+            if (answeredQuestionRatio == null) {
+                AReloadSurveyAnsweredRatioUseCase
+                        reloadSurveyUseCase = new ReloadSurveyAnsweredRatioUseCase(this);
```

Comment:
@manuelplazaspalacio  an UseCase can not be invoked from data layer, Use case should be invoked only from presentation layer. On this way database model has more of one resposability. A possible option is from where we invoke getAnsweredQuestionRatio actually inkove ReloadSurveyAnsweredRatioUseCase